shadows/framework/src/main/java/org/robolectric/shadows/ShadowGLSurfaceView.java

```diff
@@ -12,4 +16,28 @@ protected void onPause() {}
 
   @Implementation
   protected void onResume() {}
+
+  @Resetter
+  public static void reset() {
+    try {
+      // Use reflection to access the private mGLThread field
+      Field glThreadField = GLSurfaceView.class.getDeclaredField("mGLThread");
```

Comment:
> https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/opengl/java/android/opengl/GLSurfaceView.java;drc=b38c19d3b295e82ca28123dbba9d1164388988fe;l=1934?q=GLSurfaceView&ss=android
> 
> The `mGLThread` is instance scope, not class scope. So the static resetter can't work for all scenarios.

thanks for review

Reply:
> The `mGLThread` is instance scope, not class scope. So the static resetter can't work for all scenarios.

can I modify the reset() method to handle instance variables appropriately? or should I try to terminate the thread in callOnDetachFromWindow?

Reply:
> Looks like detach from window and `finalize` will quit `mGLThread`: https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/opengl/java/android/opengl/GLSurfaceView.java;l=635?q=GLSurfaceView&ss=android. I think these APIs can be shadowed. But I recommend you comparing related code between different Android versions to check whether it can work for all Android versions that supported by Robolectric.

Hi @utzcoz could you please review?